src/BoundaryConditions/apply_flux_bcs.jl

```diff
@@ -108,9 +111,18 @@ If `west_bc.condition` is a function, the function must have the signature
 
 The same logic holds for south and bottom boundary conditions in `y`, and `z`, respectively.
 """
-@inline apply_x_west_bc!(  Gc,   west_flux::BC{<:Flux}, j, k, grid, args...) = @inbounds Gc[1, j, k] += getbc(west_flux,   j, k, grid, args...) /  Δx(1, j, k, grid)
-@inline apply_y_south_bc!( Gc,  south_flux::BC{<:Flux}, i, k, grid, args...) = @inbounds Gc[i, 1, k] += getbc(south_flux,  i, k, grid, args...) /  Δy(i, 1, k, grid)
-@inline apply_z_bottom_bc!(Gc, bottom_flux::BC{<:Flux}, i, j, grid, args...) = @inbounds Gc[i, j, 1] += getbc(bottom_flux, i, j, grid, args...) / ΔzF(i, j, 1, grid)
+@inline apply_x_west_bc!(Gc, loc, west_flux::BC{<:Flux}, j, k, grid, args...) =
+    @inbounds Gc[1, j, k] += getbc(west_flux,   j, k, grid, args...) /  Δx(1, j, k, grid)
+
+@inline apply_y_south_bc!(Gc, loc, south_flux::BC{<:Flux}, i, k, grid, args...) =
+    @inbounds Gc[i, 1, k] += getbc(south_flux,  i, k, grid, args...) /  Δy(i, 1, k, grid)
```

Comment:
This won't work on a curvilinear grid, right? I'm guessing it'll have to look more like `apply_z_bottom_bc!`? But we're not setting up any problems with lateral boundary conditions yet so maybe this is fine to merge.

Reply:
Ah sorry just realized you addressed this in your original PR comment. I guess we should just be careful to not set non-zero lateral flux boundary conditions on curvilinear grids for now?

Reply:
That's correct --- I was hesitant to change it because we don't have regression tests for a case with lateral flux boundary conditions.

But thinking more about it, we do have some dynamics tests that flux boundary conditions work correctly. Maybe I should just extend those and then make this change?

Reply:
Or we can save that work for an immediate future PR.